### DISCUSSION TITLE
about - runtimeLifecycle.payouts.available() -> why does it search for empty roletokens too ?

### DISCUSSION THREAD

**Discussion by albert-kevin** [input-output-hk/marlowe-ts-sdk#151] [Q&A] [answered]
roleToken for example policyId.policyName
for example: 10137ccc04dc177e9af17c7800f3fbe86855c63d22fad293b8d98695.Party1
and somebody created a contract with empty policyId and empty name

                 (Role "")
                 (Role "")
                 (Token "" "")
                 https://preview.marlowescan.com/contractView?tab=state&contractId=71f42745070424400d0c8fb3d5df8302e53dec10d2c525c038d21c298bd14cdf%231
                 
                 when you use the SDK code runtimeLifecycle.payouts.available()
                 it searches for both your own wallet tokens and empty tokens... but the empty tokens are not from your wallet, so what about it ?
                 
                 here you see the request
                 
![image](https://github.com/input-output-hk/marlowe-ts-sdk/assets/15119906/1e8d341a-32ae-4460-ab53-a18daceb922e)

**Comment by nhenin** [comment, marked as answer]
Hi :-) Good catch :-)  

Behind the scene, the function `runtimeLifecycle.payouts.available()` : 

1. reads the asset Ids from the connected wallet as "potential" role tokens
2. calls the runtime rest client endpoint `getHeadersByRange: Payouts.GETHeadersByRange` with these "potential" role tokens 

We don't filter the `lovelaces` which have an empty policyId and an empty token name... and so this is why  you see "roleToken=.&"... It should be filtered out instead...  

FYI, I've created #153  to solve this issue.
